487 docs   expliquer le moteur de jeu (#603)
JeremieLeymarie/story-builder: documentation/docs/Builder/builder_conditional_actions.mdx

---
sidebar_position: 4
---

# Conditional actions

> Until now we have seen how we can create a story through scene creation and connection. Now we will delve further into the possibilities available to the creator to make his story more dynamic.

Reviewer: ```suggestion
Until now we have seen how we can create a story through scene creation and connection. Now we will delve further into the possibilities available to you as a creator to make your story more dynamic.
```

Conditional actions is a tool available to the user on the action scene creation pannel.

> Here we take the same story pather as in the last chapter and we create additional scenes to further enrich the story :

Reviewer: pather ? 
```suggestion
Here we take the same story path as in the last chapter and we create additional scenes to further enrich the story :
```

Author: I'm your Pather tin tin tin tin tin tin tin tin tin

<img src="/img/builder_conditional_actions.png" alt="builder scene actions" width="700" />

> Now that we have a more furnished story path we can see that the each scene has two actions that develop the story further. It is important to note that you can create more actions

Reviewer: ```suggestion
Now that we have a more furnished story path we can see that each scene has two actions that develop the story further. It is important to note that you can create more actions
```

> you are not limited to 2.

Reviewer: C'est sur une ligne à part ?

Author: Non ça c'est juste pour éviter d'avoir des lignes trop longues sur vscode mais sur la doc c'est à la suite.

> The functionnality i would like you to focus on here is linked to the little gear logo you have next to an action on the action tab of the scene creation panel.

Reviewer: ```suggestion
The feature I would like you to focus on here is linked to the little gear logo you have next to an action on the action tab of the scene creation panel.
```

> Let's say that i want to create a conditional action in my fourth scene on the seventh action :

Reviewer: ```suggestion
Let's say that I want to create a conditional action in my fourth scene on the seventh action :
```

<img src="/img/builder_conditional_actions_setting.png" alt="builder conditional actions setting" width="700" />

As you can see in the image above i have configured the seventh action to be displayed to the player only if he has reached the third scene, which by the design path of the story should not be possible as you can see in the first image of this page.

> Nonetheless if we still try and play it in the library :

Reviewer: attention au genre plz

```suggestion
As you can see in the image above I have configured the seventh action to be displayed to the player only if they have reached the third scene, which by the design path of the story should not be possible as you can see in the first image of this page.
Nonetheless if we still try and play it in the library :
```

<img src="/img/builder_conditional_actions_setting_scene1.png" alt="builder conditional actions setting scene 1" width="700" />
<img src="/img/builder_conditional_actions_setting_scene2.png" alt="builder conditional actions setting scene 2" width="700" />
<img src="/img/builder_conditional_actions_setting_scene4.png" alt="builder conditional actions setting scene 4" width="700" />

> As you can see the seventh action is hidden by questionmarks and tou can't click it because you haven't visited said third scene.

Reviewer: ```suggestion
As you can see the seventh action is hidden by question marks and you can't click it because you haven't visited said third scene.
```

> On the opposite if i change the conditional action and set it as "If played did not visit" my third scene.

Reviewer: ```suggestion
On the opposite if I change the conditional action and set it as "If played did not visit" my third scene.
```

<img src="/img/builder_conditional_actions_setting_opposite.png" alt="builder conditional actions setting opposite" width="700" />

> The seventh scene appears because we have in fact not visited the third scene :

Reviewer: pas d'espace en anglais ici

```suggestion
The seventh scene appears because we have in fact not visited the third scene:
```

<img src="/img/builder_conditional_actions_setting_opposite1.png" alt="builder conditional actions setting opposite 1" width="700" />

> You can build a really dynamic story path by granting access or not to certains scenes which creates "conditions" for the player to further explore the story to reach hidden scenes or reach the end.

Reviewer: ```suggestion
You can build a really dynamic story path by granting access or not to certain scenes which creates "conditions" for the player to further explore the story to reach hidden scenes or reach the end.
```
